src/codemirror/widget.ts

```diff
@@ -62,9 +76,34 @@ class CodeMirrorWidget extends Widget {
    }
 
   /**
+   * Handle the DOM events for the widget.
+   *
+   * @param event - The DOM event sent to the widget.
+   *
+   * #### Notes
+   * This method implements the DOM `EventListener` interface and is
+   * called in response to events on the notebook panel's node. It should
+   * not be called directly by user code.
+   */
+  handleEvent(event: Event): void {
+    switch (event.type) {
+    case 'focus':
+      this._evtFocus(event as MouseEvent);
+      break;
+    case 'blur':
+      this._evtBlur(event as MouseEvent);
```

Comment:
Should also be FocusEvent: https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L2455